Create Launch Screen Views

Vielheim opened the pull request:
Set up views for Launch, Log In and Registration

Set up custom TextField and SecureField (for consistent styling between the launch views)

Merged from main branch, resolved the merge conflicts in project.pbxproj

Commit:
Add Launch View without functionalities

Commit:
Merge branch 'main' into launch-screen-views

# Conflicts:
#	TourMate/TourMate.xcodeproj/project.pbxproj
#	TourMate/TourMate/TourMateApp.swift

Commit:
Add Launch Views

- LaunchView
- RegisterView
- LogInView

Created TitleView

Used as the main title across the Launch Views

Created custom TextFields:
- InputTextField
- InputSecureField

They mainly come with some custom styling so they'll be consistent across Log In and Register pages

Commit:
Remove unused view

Vielheim commented:
I think I created two issues on the Project board by accident. Linked them together, will just resolve when we merge

Commit:
Add Authentication Controller

In Login/Register Views, they will disable the page so that the user cannot spam click the Buttons
- As of now, unable to disable the text fields and buttons visuals (keeps triggering state updates for some reason)
- Disabled the button action being triggered
- Added spinner and warning messages

After the buttons are clicked, they will send the information to the AuthenticationController to handle

Commit:
Update username to email in log in/register pages

keanecjy commented on TourMate/TourMate/Launch/RegisterView.swift:
```diff
@@ -0,0 +1,117 @@
+
+                    InputSecureField(title: "Confirm Password *", textField: $confirmPassword)
+
+                    if showWarning {
+                        Text(warningMessage)
+                            .foregroundColor(.red)
+                            .padding()
+                    }
+
```
I think the warning should be below, otherwise it shifts the above fields up

Vielheim replied:
Below the Button?

keanecjy replied:
yes

Vielheim replied:
Ok! I shifted it

keanecjy reviewed (approved):
Looks good, small nit to fix

Commit:
Update UI when pressing Submit

Disables the TextFields and Buttons, set their opacity to 0.5 to denote disabled

Commit:
Update warning message location
